python/cuml/cuml/linear_model/logistic_regression_mg.pyx

```diff
@@ -255,11 +257,24 @@ class LogisticRegressionMG(MGFitMixin, LogisticRegression):
 
     def fit(self, input_data, n_rows, n_cols, parts_rank_size, rank, convert_dtype=False):
 
-        assert len(input_data) == 1, f"Currently support only one (X, y) pair in the list. Received {len(input_data)} pairs."
```

Comment:
Do you also think that using an `assert` for this is a bug? For me it is because there are cases when `assert` statements are not executed by CPython, and we always want this to be checked.